### ISSUE TITLE
Relationship strength "composite" has no effect in the app

### ISSUE THREAD

**Issue by Ojaybee** [MerginMaps/mobile#2315]
Setting a 1-n relationship in QGIS is saved in the .qgz project file. There is an option to set the strength of the relationship as shown in the [Mergin docs](https://merginmaps.com/docs/layer/one-to-n-relations/). Using the Association setting allows deletion of the parent while child data persists. Changing this to Composition allows for cascading deletes in QGIS (if a parent is deleted child items are also deleted). QGIS will pop up a warning as to which child items from which table will be deleted.

This behaviour is not the same in Input. Regardless of relationship strength setting, deleting a parent feature will not delete child items. This leads to orphans in the related table.

**Comment by jfdarveau** [comment]
I just encountered the same issue and I would like to add that when adding a child feature into a form, it seems to be saved immediately regardless of the fact that the parent feature hasn't been saved yet. Cancelling the parent form doesn't delete the child feature and the foreign key pointing to the nonexistent parent feature is still present in the attribute table. I'm my opinion, this problem should be taken seriously because it may leave many orphan objects into child tables.

**Comment by cmparada** [comment]
I was searching and found this thread, I'm encountering the same issue in Mergin Maps. The child features are easily deleted in QGIS but deleting a parent feature within Mergin Maps leads to orphaned child features. Is there a work around for this? I've experimented with polymorphic relations with no success.

**Comment by tomasMizera** [comment]
Hi @tmcallahan59 - thanks for the comment, I've reclassified the ticket. Do you prefer using `Composition` strength instead of the default one (association)?

**Comment by tmcallahan59** [comment]
Yes, I generally prefer Composition as the default as it better fits the tightly coupled relationships I usually work with. That said, Association makes sense in cases like referencing shared features, e.g., linking monitoring stations across multiple sampling campaigns without implying ownership.

**Comment by cmparada** [comment]
I prefer a Composition strength as well. Thank you for your comment. I hadn't thought to run a script to delete orphans, but my forms are set up for field technicians and I'm afraid of deleting data that would possibly be important. If this was fixed it would be a great fail safe for the other reliabilities I have in our field collection forms.
